src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/SingleMethodCompilationModuleGroup.cs

```diff
@@ -12,16 +13,20 @@ namespace ILCompiler
     /// </summary>
     public class SingleMethodCompilationModuleGroup : CompilationModuleGroup
     {
+        private ModuleDesc _module;
         private MethodDesc _method;
 
-        public SingleMethodCompilationModuleGroup(MethodDesc method)
+        public SingleMethodCompilationModuleGroup(ModuleDesc module, MethodDesc method)
         {
+            _module = module;
             _method = method;
         }
 
+        public override IList<ModuleDesc> CompilationModules => new[] { _module };
```

Comment:
The problem with exposing a list of modules is that it doesn't map well to special compilation more like this one (we don't know what the consumer will do with this list). I've been trying to avoid exposing that for that reason (and expose higher level concepts instead). Not blocking, but something to think about.